examples/rust/publish_subscribe_cross_language/README.md

```diff
@@ -0,0 +1,132 @@
+> compatible with shared memory. See the
+> [complex data type example](../complex_data_types) for guidance on how to
+> use them.
+
+This example illustrates a robust publisher-subscriber communication pattern
+between a Rust publisher and a C++ subcriber application. The Rust publisher
+application sends a message every second, each containing `TransmissionData` and
+the `CustomHeader`. On the receiving end, the C++ subscriber application prints
+the received payload and the user header to the console whenever new data
+arrives.
+
+First you have to build the C++ examples:
+
+```sh
+cmake -S . -B target/ffi/build -DBUILD_EXAMPLES=ON
+cmake --build target/ffi/build
+```
+
+To observe this dynamic communication in action, open two separate terminals and
+execute the following commands:
+
+### Terminal 1
+
+Run the C++ subscriber application from the
+[C++ cross-language publish-subcribe example](../../cxx/publish_subscribe_cross_language):
+
+```sh
+./target/ffi/build/examples/cxx/publish_subscribe_cross_language/example_cxx_publish_subscribe_cross_language_subscriber
+```
+
+### Terminal 2
+
+Run the Rust publisher application:
+
+```sh
+cargo run --example publish_subscribe_cross_language_publisher
+```
+
+Feel free to also run the Rust subscriber and the C++ publisher applications
+simultaneously to explore how iceoryx2 handles cross-language
+publisher-subscriber communication efficiently.
+
+### Terminal 3
+
+Run the Rust subscriber application:
+
+```sh
+cargo run --example publish_subscribe_cross_language_subscriber
+```
+
+### Terminal 4
+
+Run the C++ publisher application from the
+[C++ cross-language publish-subscribe example](../../cxx/publish_subscribe_cross_language):
+
+```sh
+./target/ffi/build/examples/cxx/publish_subscribe_cross_language/example_cxx_publish_subscribe_cross_language_publisher
+```
+
+You can also communicate with the C publisher and subscriber applications from
+the
+[C cross-language publish-subscribe example](../../c/publish_subscribe_cross_language).
+
+You may hit the maximum supported number of ports when too many publisher or
+subscriber processes are running. Check the [iceoryx2 config](../../../config)
+to set the limits globally or refer to the
+[API of the Service builder](https://docs.rs/iceoryx2/latest/iceoryx2/service/index.html)
+to set them for a single service.
+
+## How to enable publish-subscribe communication between Rust, C++ and C applications
+
+To communicate with each other, publisher and subscriber applications must share
+the same service configuration, including the payload and the user header type
+name.
+
+For C++ and Rust applications, the internally derived type names
+usually depend on the used programming language. To allow cross-language
+communication involving C++ applications, iceoryx2 provides the possibility to
+customize the payload and the user header type name by setting `TYPE_NAME` in
+the sent C++ data struct and user header, e.g.
+
+```cxx
```

Comment:
Please focus in the readme only on the specific language configuration and remove all the other stuff.

Otherwise we have to add in every readme in the future also an explanation how to configure python, go, lua etc. 
If someone wants to know this, they just need to open the language specific example.